File: 1501-2000/1528.Shuffle String.py
class Solution(object):
    def restoreString(self, s, indices):
        """
        :type s: str
        :type indices: List[int]
        :rtype: str
        """
        res = [None] * len(s)
        for i, target in enumerate(indices):
            res[target] = s[i]
        return "".join(res)

File: 1501-2000/1528.Shuffle String.c
char * restoreString(char * s, int* indices, int indicesSize){
    for (int i = 0; i < indicesSize; i++) {
        int j = i;
        while (indices[j] >= 0)
        {
            char tmp = s[i]; s[i] = s[indices[j]]; s[indices[j]] = tmp;
            indices[j] = ~indices[j];
            j = ~indices[j];
        }
    }
    return s;
}